JenkinsMaster.groovy

```diff
@@ -0,0 +1,91 @@
+pipeline {
+    agent {
+        label 'docker01'
+    }
+    options {
+        ansiColor colorMapName: 'XTerm'
+    }
+    parameters {
+        string(
+            name: 'APP_SERVERS',
+            defaultValue: '10.10.10.122',
+            description: 'Deploy container to these servers. List of servers separated by comma.'
+        )
+        booleanParam(
+            name: "BUILD_IMAGE",
+            defaultValue: true,
+            description: "Build image and upload it to Docker registry"
+        )
+    }
+    stages {
+        when {
+            expression {
+                return params.BUILD_IMAGE
+            }
+        }
+        stage('Build') {
+            steps {
+                sh "docker build --rm=true -t captain-hook-master ."
+            }
+        }
+
+        stage('Prepare and upload to registry ') {
+            when {
+                expression {
+                    return params.BUILD_IMAGE
+                }
+            }
+            steps {
+                withCredentials([string(credentialsId: 'docker-registry-azure', variable: 'DRpass')]) {
+                    sh 'docker login roihunter.azurecr.io -u roihunter -p "$DRpass"'
+                    sh "docker tag captain-hook-master roihunter.azurecr.io/captain-hook/master"
+                    sh "docker push roihunter.azurecr.io/captain-hook/master"
+                    sh "docker rmi captain-hook-master"
+                    sh "docker rmi roihunter.azurecr.io/captain-hook/master"
+                }
+            }
+        }
+
+        stage('Deploy API container') {
+            steps {
+                withCredentials([
+                    string(credentialsId: 'docker-registry-azure', variable: 'DRpass')
+                ]) {
+                    script {
+                        def servers = params['app_servers'].tokenize(',')
```

Comment:
@Joozty This parameter should be upper-cased since you declared uppercase variable name above.